test/bandit/headers_test.exs

```diff
@@ -41,4 +41,40 @@ defmodule Bandit.HeadersTest do
       end
     end
   end
+
+  describe "parse_integer/1" do
+    @non_neg_integers [0, 1, 420, 999_999_999]
```

Comment:
we should also test leading zeroes, as they're explicitly called out as supported in the spec